src/js_notifications.hpp

```diff
@@ -0,0 +1,163 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <map>
+#include <vector>
+
+#include <realm/object-store/collection_notifications.hpp>
+
+#include "js_types.hpp"
+
+namespace realm::js::notifications {
+
+using IdType = unsigned long long;
```

Comment:
Why not `size_t`?

Reply:
@fronck can answer to this: I believe this was to prevent a very small range if we started publishing for more embedded devices.

Reply:
`size_t` is the natural pointer size on a system.  Is it not guaranteed to be any larger than 64k.  So, just in the interest of not getting nasty surprises if we were to, for example, get into the IoT space, I wanted to force it to hold more than that.